Make the interface to be compatible with GCL-compiled fricas package

cxzhong opened the pull request:
<!-- ^ Please provide a concise and informative title. -->
<!-- ^ Don't put issue numbers in the title, do this in the PR description below. -->
<!-- ^ For example, instead of "Fixes #12345" use "Introduce new method to calculate 1 + 2". -->
<!-- v Describe your changes below in detail. -->
<!-- v Why is this change required? What problem does it solve? -->
<!-- v If this PR resolves an open issue, please link to it here. For example, "Fixes #12345". -->
Most people use the Ubuntu or Debian system. But they do not provide a compatible fricas compiled by sbcl. So I fix it with gcl's fricas compatibility.
Fix #40569 


### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->

- [ ] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### :hourglass: Dependencies

<!-- List all open PRs that this PR logically depends on. For example, -->
<!-- - #12345: short description why this is a dependency -->
<!-- - #34567: ... -->

cxzhong committed:
Help the interface to be compatible with Ubuntu's fricas

cxzhong committed:
Update fricas.py

cxzhong committed:
fix lint

cxzhong committed:
Improve speed in ecl's fricas

cxzhong committed:
Remove unnecessary error check

dimpase reviewed (changes requested):
simplify Lisp detection

cxzhong commented on src/sage/interfaces/fricas.py:
```diff
@@ -353,6 +355,28 @@ def _start(self):
+                E.expect(self._prompt, timeout=0.1)
+            except pexpect.TIMEOUT:
+                break
+        E.sendline(')lisp (lisp-implementation-type)')
```
@dimpase in this line I ask that lisp-implementation-type

cxzhong replied:
And there are other message like 
```
this Function declaration sageprint : InputForm -> String has been added to workspace.
```
When the fricas running.
So we need to clean this.

dimpase replied:
```
# GCL-based FriCAS may output function declaration messages
# asynchronously after the prompt appears.
```
Maybe there is a FriCAS way to suppress these?
Please ask on the FriCAS forum, before starting to mess around with all these timeouts, which are really a weapon of last resort.

cxzhong replied:
Yes, we need to ask this

cxzhong replied:
This function is added when init fricas. But sbcl version of fricas does not output that.

dimpase commented on src/sage/interfaces/fricas.py:
```diff
@@ -353,6 +355,28 @@ def _start(self):
+        E = self._expect
+        # First, consume any buffered output from previous commands
+        # to ensure we get a clean response to the lisp command
+        while True:
```
What's going on in this loop then?

cxzhong replied:
I do that for all other messages is clean. So we can run this list-umplementation-type safely

cxzhong committed:
fix a synatx issue in gcl fricas

cxzhong committed:
revert

cxzhong committed:
Fix FriCAS command execution for proper quitting

Updated command execution for FriCAS to use stdin for quitting instead of -eval to avoid misinterpretation by GCL-compiled FriCAS.

cxzhong committed:
Decrease timeout for fricas subprocess command

Reduce timeout for subprocess call to fricas from 30 to 1 second.

cxzhong committed:
Refactor subprocess command execution in fricas.py

Refactor subprocess call to avoid timeout handling.

cxzhong committed:
Merge branch 'sagemath:develop' into patch-10
